openedx/core/djangoapps/util/management/commands/run_aggregator.py

```diff
@@ -0,0 +1,272 @@
+            self.course_breadth, chapter_count, sequential_count, vertical_count
+        )
+        print u"Learners: {}".format(self.learners_count)
+        print u"Completions: {}".format(self.completions_count)
+
+        query_data = {}
+        for query in self.executed_queries:
+            query_type = query['sql'].split()[0]
+            query_data.setdefault(query_type, {'count': 0, 'times': []})
+            query_data[query_type]['count'] = query_data[query_type]['count'] + 1
+            query_data[query_type]['times'].append(float(query['time']))
+
+        print u"SQL Queries | All Count: {}".format(len(self.executed_queries))
+        for query_type, data in query_data.items():
+            print u"SQL Queries | {} Count: {} Time: {} Percentiles: {}".format(
+                query_type, data['count'], sum(data['times']), self._get_percentiles(data['times'])
+            )
+
+        if time_taken:
+            print u"Total Time: {:.3f}s".format(time_taken)
+
+    def _print_results_footer(self):
+        """ Print footer. """
+        print u"----------------------------------------------------------"
+
+    def _get_percentiles(self, items):
+        return " | ".join(
+            [u"{}%: {:.3f}s".format(p, numpy.percentile(items, p)) for p in [
+                50, 66, 75, 80, 90, 95, 98, 99, 100]
+             ]
+        )
+
+    def _copy_executed_queries(self):
+        self.executed_queries = list(connection.queries_log)
+
+    def _update_aggregators(self, username, course_key, block_keys=(), force=False):
+        from completion_aggregator.tasks import update_aggregators
+        update_aggregators(username, course_key, block_keys=block_keys, force=force)
+
+    def _complete_blocks_for_users(self, blocks, users):
+        for user in users:
+            for block in blocks:
+                BlockCompletion.objects.submit_completion(
+                    user=user,
+                    course_key=self.course.id,
+                    block_key=block.location,
+                    completion=1.0
+                )
+                self._update_aggregators(user.username, unicode(self.course.id), block_keys=[unicode(block.location)])
+
+    def _assert_vertical_completion_for_all_users(self, vertical, expected_completion):
+        from completion_aggregator.models import Aggregator
+        for user in self.users:
+            vertical_completion = Aggregator.objects.get(
+                user=user, course_key=self.course.id, block_key=vertical.location
+            ).percent
+            assert abs(vertical_completion - expected_completion) < 0.01
+
+    def _time_handler(self, handler, **kwargs):
+        """ Time how long it takes to run handler. """
+        timer_start = self.timer()
+        handler(**kwargs)
+        timer_end = self.timer()
+        return timer_end - timer_start
+
+    def test_course_published_handler_when_block_is_added(self):
+
```

Comment:
These course-modification tests should populate a good number of BlockCompletions before modifying the course. (But only evaluate the course modification).

Reply:
Sounds like a good idea. Will add this.